### PR TITLE
[JIT] Add SchemaCheckMode OpInfo test

### PR DESCRIPTION
**Pull request by ttanpcs**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82442
- Move test_schema_check to torch/test directory.
- Add opInfo test for SchemaCheckMode to check all operator schemas 
- Add various changes (using isClose instead of equals, skipping complex number cases for certain ops, etc...) in order to have test_schema_check pass.

Differential Revision: [D38437946](https://our.internmc.facebook.com/intern/diff/D38437946)

### CODE REVIEW

File: torch/csrc/jit/python/init.cpp [hunk 1]
```diff
@@ -233,10 +233,12 @@ bool loadPythonClasses() {
...
-bool isEmptyContainer(const py::handle self) {
-  bool is_empty_list =
-      PySequence_Check(self.ptr()) && !PySequence_Size(self.ptr());
-  return is_empty_list;
+c10::optional<IValue> toTypeInferredIValueOptional(py::handle input) {
+  try {
+    return toTypeInferredIValue(input);
+  } catch (const c10::Error& e) {
+    return c10::nullopt;
+  }
```
Comment (davidberard98) [inline torch/csrc/jit/python/init.cpp:241]: could you add a comment why we need to catch errors here

File: torch/csrc/jit/python/init.cpp [hunk 2]
```diff
@@ -1879,16 +1882,21 @@ void initJITBindings(PyObject* module) {
               }),
           py::call_guard<py::gil_scoped_release>());
   m.def("_is_alias_of", [](const py::object& self, const py::object& other) {
-    if (isEmptyContainer(self) || isEmptyContainer(other)) {
+    c10::optional<IValue> self_value = toTypeInferredIValueOptional(self);
+    c10::optional<IValue> other_value = toTypeInferredIValueOptional(other);
+
+    if (!self_value || !other_value) {
```
Comment (davidberard98) [inline torch/csrc/jit/python/init.cpp:1888]: maybe also add a comment here (e.g. "only return true if we can be certain that they are aliasing")

**Review by davidberard98 — COMMENTED** [review]
looks good, but can you add some comments? lots of special cases here, would be good to document what errors led to these special cases.

**Comment by goldenxuett** [comment]
@goldenxuett has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38437946).

**Review by davidberard98 — APPROVED** [review]
Looks good!! Remember to land via internal workflow since there are internal changes.

**Comment by goldenxuett** [comment]
@goldenxuett has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38437946).

**Pull request closed without merging**